src/map/path.cpp

```diff
@@ -431,46 +431,37 @@ bool path_search(struct walkpath_data *wpd, int16 m, int16 x0, int16 y0, int16 x
 	return false;
 }
 
-
 //Distance functions, taken from http://www.flipcode.com/articles/article_fastdistance.shtml
 bool check_distance(int dx, int dy, int distance)
 {
 #ifdef CIRCULAR_AREA
-	//In this case, we just do a square comparison. Add 1 tile grace for diagonal range checks.
-	return (dx*dx + dy*dy <= distance*distance + (dx&&dy?1:0));
+	dx = std::abs(dx);
```

Comment:
This is not really needed because if you if multiply the value with itself you will always have a positive result.
We also shouldn't remove comments.

Reply:
i  put it here as mistake, mybad